File: content/tracks/dsa/problems/lc-0167-two-sum-ii-input-array-is-sorted/Solution.java
import java.util.*;

class Solution {
    public int[] twoSum(int[] numbers, int target) {
        // The array is sorted: a sum that is too small moves the left end up, too large moves the
        // right end down. The answer is 1-indexed.
        int left = 0;
        int right = numbers.length - 1;
        while (left < right) {
            int total = numbers[left] + numbers[right];
            if (total == target) {
                return new int[] {left + 1, right + 1};
            }
            if (total < target) {
                left++;
            } else {
                right--;
            }
        }
        return new int[0];
    }
}

File: content/tracks/dsa/problems/lc-0167-two-sum-ii-input-array-is-sorted/solution.go
package main

func twoSum(numbers []int, target int) []int {
	// The array is sorted: a sum that is too small moves the left end up, too large moves the
	// right end down. The answer is 1-indexed.
	left, right := 0, len(numbers)-1
	for left < right {
		total := numbers[left] + numbers[right]
		if total == target {
			return []int{left + 1, right + 1}
		}
		if total < target {
			left++
		} else {
			right--
		}
	}
	return []int{}
}
